File: leetcode/lc-us/cpp/605.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    bool canPlaceFlowers(vector<int> &flowerbed, int n) {
        flowerbed.insert(flowerbed.begin(), 0);
        flowerbed.push_back(0);
        for (int i = 1; i < flowerbed.size() - 1; i++) {
            if (flowerbed[i] == 0 && flowerbed[i - 1] == 0 && flowerbed[i + 1] == 0) {
                flowerbed[i] = 1;
                n--;
            }
            if (n <= 0) {
                return true;
            }
        }
        return false;
    }
};

File: leetcode/lc-us/rust/605.rs
impl Solution {
    pub fn can_place_flowers(flowerbed: Vec<i32>, n: i32) -> bool {
        let m = flowerbed.len();
        let mut cnt = 0;
        let mut flag = false;
        for i in 0..m {
            if flowerbed[i] == 0 {
                if !flag && (i == m - 1 || flowerbed[i + 1] == 0) {
                    cnt += 1;
                    flag = true;
                } else {
                    flag = false;
                }
            } else {
                flag = true;
            }
        }
        return cnt >= n;
    }
}